Content.Client/UserInterface/LateJoinGui.cs

```diff
@@ -0,0 +1,116 @@
+using Robust.Client.Console;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.Utility;
+using Content.Shared.Jobs;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Log;
+using Robust.Shared.Maths;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+
+namespace Robust.Client.UserInterface.CustomControls
+{
+    public sealed class LateJoinGui : SS14Window
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IPrototypeManager _prototypeManager;
+        [Dependency] private readonly IClientConsole _console;
+#pragma warning restore 649
+
+        protected override Vector2? CustomSize => (360, 560);
+
+        public event Action<string> SelectedId;
+
+        public LateJoinGui()
+        {
+            IoCManager.InjectDependencies(this);
+
+            Title = Loc.GetString("Late Join");
+
+            var jobList = new VBoxContainer();
+            var vBox = new VBoxContainer
+            {
+                    Children =
+                    {
+                        new ScrollContainer
+                        {
+                            SizeFlagsVertical = SizeFlags.FillExpand,
+                            Children =
+                            {
+                                jobList
+                            }
+                        }
+                    }
+                };
+            Contents.AddChild(vBox);
+
+            foreach (var job in _prototypeManager.EnumeratePrototypes<JobPrototype>().OrderBy(j => j.Name))
```

Comment:
Is it intentional to show all jobs here? even if they're not available?

Might want to do 1 AvailablePositions() call before this loop, and then filter out jobs here using .ContainsKey() on the available dict.

Reply:
See discord.

Reply:
Ok. be about half an hour/forty five mins cos traffic.